pkg/controllers/dynakube/istio/reconciler.go

```diff
@@ -22,6 +23,7 @@ type Reconciler interface {
 	ReconcileAPIUrl(ctx context.Context, dynakube *dynatracev1beta2.DynaKube) error
 	ReconcileCodeModuleCommunicationHosts(ctx context.Context, dynakube *dynatracev1beta2.DynaKube) error
 	ReconcileActiveGateCommunicationHosts(ctx context.Context, dynakube *dynatracev1beta2.DynaKube) error
+	ReconcileCSIDriver(ctx context.Context, dynakube *dynatracev1beta2.DynaKube) error
```

Comment:
`ReconcileCSIDriver` should always be called after `ReconcileCodeModuleCommunicationHosts` as far as I got it.
I would call `ReconcileCSIDriver` then directly within `ReconcileCodeModuleCommunicationHosts` instead of creating an additional interface function.

Reply:
I can happily do that, there is only one question: Wouldn't this lead to `ReconcileCodeModuleCommunicationHosts` as a method doing too many things? Or can the `host` of the `codeModulesImage` URL be considered as a communication host?

also maybe interesting for you @0sewa0

Reply:
> Or can the host of the codeModulesImage URL be considered as a communication host?

that is what you are doing here: `err = r.reconcileCommunicationHosts(ctx, []dtclient.CommunicationHost{codeModulesHost}, CSIDriverComponent)`

the fact that the `istio.Reconciler` works with `dtclient.CommunicationHost` is just for historical reasons, there is no better reason for it, it was just always used for creating `ServiceEntries` for a given list of `dtclient.CommunicationHost`s

I like the idea that we don't bloat the (hacky) interface, and just do both in the `ReconcileCodeModuleCommunicationHosts`, but you would probably want to rename that function to something "more correct". Like `ReconcileInjectionEndpoints` or something smarter

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/3343/commits/fdc1c92c54d8f068b0b02b3b4637cdaf7048034e